artifact/sha/sha_test.cpp

```diff
@@ -34,9 +34,9 @@ TEST(ShasummerTest, TestShaSum) {
 
 	vector<uint8_t> actual(4096);
 
-	auto bytes_read = r.Read(actual);
+	auto bytes_read = r.Read(actual.begin(), actual.end());
 
-	EXPECT_TRUE(bytes_read);
+	ASSERT_TRUE(bytes_read) << bytes_read.error();
```

Comment:
Nit: doing more than the commit says.

Reply:
Oops, yes, that wasn't even supposed to be there (I'm surprised it works actually. I thought `error()` would throw in that context).

Reply:
Hehe, I wish. I believe only `&&` and `||` are eligible for lazy evaluation (short circuiting, as I think it's called in C++).